src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -512,11 +522,11 @@ namespace OpenLoco::World::MapGenerator
                     continue;
                 }
                 const auto typePattern = S5::getOptions().landDistributionPatterns[landObjectIdx];
-                if (typePattern != pattern)
+                if (typePattern != landDistributionPatterns[i])
                 {
                     continue;
                 }
-                _generateFuncs[enumValue(pattern)](heightMap, landObjectIdx);
+                _generateFuncs[i](heightMap, landObjectIdx);
```

Comment:
this will call the wrong function. It needs to be the `enumValue(landDistributionPatterns[i])`